xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/webjar/Logic.js

```diff
@@ -59,520 +67,763 @@ define(["jquery", "polyfills"], function ($) {
+     * @param {String} property The property to sort according to
+     * @param {String} level The sort level of the property (0 is the highest).
+     *   Undefined means current. Negative value removes property sort.
+     * @param {String} descending Specify whether the sort should be descending or not.
+     *   Undefined means toggle current direction
+     */
+    sort: function (property, level, descending) {
+      if (!this.isValidPropertyId(property)) { return; }
+      if (!this.isPropertySortable(property)) { return; }
+      // find property current sort level
+      var currentLevel = this.data.query.sort.findIndex(function (sortObject) {
+        return sortObject.property === property;
+      });
+      // default level
+      if (level === undefined) {
+        level = (currentLevel !== -1) ? currentLevel : 0;
+      } else if (level < 0) {
+        level = -1;
+      }
+      // default descending
+      if (descending === undefined) {
+        descending = (currentLevel !== -1) ? !this.data.query.sort[currentLevel].descending : false;
+      }
+      // create sort object
+      var sortObject = {
+        property: property,
+        descending: descending,
+      };
+      // apply sort
+      if (level !== -1) {
+        this.data.query.sort.splice(level, 1, sortObject);
+      }
+      if (currentLevel !== -1 && currentLevel !== level) {
+        this.data.query.sort.splice(currentLevel, 1);
+      }
+      // dispatch events
+      this.triggerEvent("sort", {
+        property: property,
+        level: level,
+        descending: descending,
+      });
+
+      // CALL FUNCTION TO FETCH NEW DATA HERE
+    },
+
+
+    /**
+     * Add new sort entry, shorthand of sort:
+     * If the property is already sorting, does nothing
+     * @param {String} property The property to add to the sort
+     * @param {String} descending Specify whether the sort should be descending or not.
+     *   Undefined means toggle current direction
+     */
+    addSort: function (property, descending) {
+      var currentLevel = -1;
+      this.data.query.sort.some(function (sortObject, i) {
+        if (sortObject.property === property) {
+          currentLevel = i;
+          return;
         }
+      });
+      if (currentLevel !== -1) { return; }
+      this.sort(property, this.data.query.sort.length, descending);
+    },
+
+
+    /**
+     * Remove a sort entry, shorthand of sort:
+     * @param {String} property The property to remove to the sort
+     */
+    removeSort: function (property) {
+      this.sort(property, -1);
+    },
+
+
+
+
+    /**
+     * ---------------------------------------------------------------
+     * FILTER
+     */
+
+
+    /**
+     * Returns whether a certain property is filterable or not
+     * @param {String} propertyId
+     * @returns {Boolean}
+     */
+    isPropertyFilterable: function (propertyId) {
+      return !!this.getFilterDescriptor(propertyId).id;
```

Comment:
We can do as you think is better. For now I was doing:
- falsy filter id -> not filterable (we enforce the value to disable filtering)
- no filter descriptor at all -> default filter (we do not specify, and fallback to default filter)

If using `none` is more convenient for you I'm fine to change too. Let's do what is easier at the end!

Reply:
OK, so the only difference is that I was using ``none`` while you expect falsy filter id. I cannot return ``false`` (the filter id type is String on my side) but I could return ``null`` or '' (empty string). I think this is better indeed, so I'll update my code. Note that you may need to update the algorithm used to merge the filter configuration, see above. I'm also wondering whether we shouldn't do the same with displayers:

* fall back on the default displayer (when merging the configuration) if the id is undefined (missing)
* don't display the property if the displayer id is falsy

WDYT?

Reply:
I'm taking note for the filter merge, that would be too bad to have two filters being the same because they have an id equal to `null`! :)

Concerning the displayers behavior, I think that the value should be displayed whatever happens. Do you have any cases where we need a property have a falsy displayer id, so that all of the entries properties are not displayed? In that case, it would make more sense to hide the whole column (or smth else in another layout) with the visibility option. Or to even not use this column at all in the LD in the first place, as the displayers options are not modifiable by the user.

Also you said:
> fall back on the default displayer (when merging the configuration) if the id is undefined (missing)

for now I'm falling back on the typeDescriptor filter then default filter when the **filter** object in config is not defined. If there is the filter object but it has a falsy id (I'm considering `undefined` as falsy here too), then it's not filterable.